FTL v5.3.1 fails to build with gcc 10.2.x

### Platform
- OS and version: Gentoo

I'm aware that Gentoo is not a supported platform but due to the nature of the bug it might be reproducible on other platforms supporting gcc-10.2.x too. I tried to reproduce it on Fedora 32 which unfortunately failed because of a missing `libidn-static` (`libidn.a`)

### Expected behavior

FTL build completes successful.

### Actual behavior / bug
```
$ gcc --version
gcc (Gentoo 10.2.0-r3 p4) 10.2.0
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```
Error due to potential string truncation:
```
$ ./build.sh
[...]
Building Pi-hole FTL daemon                                    
   - Branch: master                                            
   - Architecture: x86_64 (compiled locally)
   - Version: v5.3.1
   - Tag: v5.3.1
   - Hash: e1db31d
   - Commit date: 2020-11-28 21:59:12 +0100
[...]
[ 50%] Building C object src/dnsmasq/CMakeFiles/dnsmasq.dir/inotify.c.o                                                                                        
[ 50%] Building C object src/dnsmasq/CMakeFiles/dnsmasq.dir/ipset.c.o          
In file included from /usr/include/string.h:519,                              
                 from /usr/include/sys/un.h:37,                                                                                                                
                 from /home/ganto/FTL/src/dnsmasq/dnsmasq.h:105,           
                 from /home/ganto/FTL/src/dnsmasq_interface.c:12:
In function 'strncpy',
    inlined from '_FTL_new_query' at /home/ganto/FTL/src/dnsmasq_interface.c:558:3:
/usr/include/bits/string_fortified.h:106:10: error: '__builtin_strncpy' output may be truncated copying 45 bytes from a string of length 45 [-Werror=stringop-truncation]
  106 |   return __builtin___strncpy_chk (__dest, __src, __len, __bos (__dest));
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[...]
```

### Steps to reproduce

If you want to reproduce it on Gentoo you can use the [feature/pi-hole](https://github.com/ganto/linuxmonk-overlay/tree/feature/pi-hole/net-dns) branch from my linuxmonk-overlay.

Otherwise I would expect any other Linux distribution that meets the requirements and a recent enough gcc version to fail with the same error. 

### Additional context

This is not a bug that affects the end users but sooner or later you might step over it too. Note that the same code compiles fine when using `gcc-9.3.0` on Gentoo. Also FTL v5.2 compiled fine with `gcc-10.2.0`.

I just tested with the `feature/v5.3.2` branch but it seems this issue is not (yet) resolved:
```
$ git checkout release/v5.3.2                                                                                                               
Branch 'release/v5.3.2' set up to track remote branch 'release/v5.3.2' from 'origin'. 
Switched to a new branch 'release/v5.3.2'                                  
$ ./build.sh                                                                                                                                
-- The C compiler identification is GNU 10.2.0                               
-- Detecting C compiler ABI info                                             
-- Detecting C compiler ABI info - done                                    
-- Check for working C compiler: /usr/bin/cc - skipped                  
-- Detecting C compile features
-- Detecting C compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Building FTL with readline support: NO
-- Embedded LUA will use readline for history: NO
-- Configuring done
-- Generating done
-- Build files have been written to: /home/ganto/FTL/cmake
[...]
Building Pi-hole FTL daemon
   - Branch: release/v5.3.2
   - Architecture: x86_64 (compiled locally)
   - Version: v5.3.1-8-gd25f05d
   - Tag: v5.3.1
   - Hash: d25f05d
   - Commit date: 2020-11-29 13:21:24 +0100
[...]
 47%] Building C object src/dnsmasq/CMakeFiles/dnsmasq.dir/domain.c.o
[ 47%] Built target tre-regex
[ 47%] Building C object src/dnsmasq/CMakeFiles/dnsmasq.dir/dump.c.o
[ 47%] Building C object src/CMakeFiles/FTL.dir/gc.c.o
In file included from /usr/include/string.h:519,
                 from /usr/include/sys/un.h:37,
                 from /home/ganto/FTL/src/dnsmasq/dnsmasq.h:105,
                 from /home/ganto/FTL/src/dnsmasq_interface.c:12:
In function 'strncpy',
    inlined from '_FTL_new_query' at /home/ganto/FTL/src/dnsmasq_interface.c:558:3:
/usr/include/bits/string_fortified.h:106:10: error: '__builtin_strncpy' output may be truncated copying 45 bytes from a string of length 45 [-Werror=stringop-truncation]
  106 |   return __builtin___strncpy_chk (__dest, __src, __len, __bos (__dest));
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/src/dnsmasq_interface.c
+++ b/src/dnsmasq_interface.c
@@ -551,12 +551,12 @@ bool _FTL_new_query(const unsigned int flags, const char *name,
 	// 127.0.0.1 to avoid queries originating from localhost of the
 	// *distant* machine as queries coming from the *local* machine
 	const sa_family_t family = (flags & F_IPV4) ? AF_INET : AF_INET6;
-	char clientIP[ADDRSTRLEN] = { 0 };
+	char clientIP[ADDRSTRLEN+1] = { 0 };
 	if(config.edns0_ecs && edns->client_set)
 	{
 		// Use ECS provided client
 		strncpy(clientIP, edns->client, ADDRSTRLEN);
-		clientIP[ADDRSTRLEN-1] = '\0';
+		clientIP[ADDRSTRLEN] = '\0';
 	}
 	else
 	{

